File: service/src/main/java/mu/server/service/service/AuthUserDetailsService.java
package mu.server.service.service;

import org.jspecify.annotations.NonNull;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

public interface AuthUserDetailsService extends UserDetailsService {
    @Override
    @NonNull UserDetails loadUserByUsername(@NonNull String username) throws UsernameNotFoundException;
}

File: service/src/main/java/mu/server/service/service/AuthUserDetailsService.kt
package mu.server.service.service

import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService

@Deprecated(" Using keycloak")
interface AuthUserDetailsService : UserDetailsService {

    override fun loadUserByUsername(username: String): UserDetails
}